src/RA_AchievementSet.cpp

```diff
@@ -420,148 +423,142 @@ BOOL AchievementSet::LoadFromFile(GameID nGameID)
     const std::string sFilename = GetAchievementSetFilename(nGameID);
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pFile = nullptr;
-    errno_t nErr = fopen_s(&pFile, sFilename.c_str(), "r");
-    if (pFile != nullptr)
-    {
-        //	Store this: we are now assuming this is the correct checksum if we have a file for it
-        g_pCurrentGameData->SetGameID(nGameID);
+    std::ifstream ifile{ sFilename };
 
-        if (m_nSetType == Local)
-        {
-            const char EndLine = '\n';
-            char buffer[4096];
-            unsigned long nCharsRead = 0;
+    // in this case we do have to check
+    if (!ifile.is_open())
+        return FALSE;
 
-            //	Get min ver:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-            //	UNUSED at this point? TBD
 
-            //	Get game title:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-            if (nCharsRead > 0)
-            {
-                buffer[nCharsRead - 1] = '\0';	//	Turn that endline into an end-string
+    //	Store this: we are now assuming this is the correct checksum if we have a file for it
+    g_pCurrentGameData->SetGameID(nGameID);
 
-                //	Loading Local from file...?
-                g_pCurrentGameData->SetGameTitle(buffer);
-            }
-
-            while (!feof(pFile))
+    if (m_nSetType == Local)
+    {
+        std::string buffer;
+        buffer.reserve(4096); 
+
+        //	Get min ver:
+        ifile.getline(buffer.data(), buffer.capacity());
+        auto nCharsRead{ ifile.gcount() };  //	UNUSED at this point? TBD 
+                  
+        //	Get game title:
+        ifile.getline(buffer.data(), buffer.capacity());
+        nCharsRead = ifile.gcount();
+        //	Loading Local from file...?
+        if (nCharsRead > 0)
+            g_pCurrentGameData->SetGameTitle(buffer.data());
+
+        while (!ifile.eof())
+        {
+            ifile.getline(buffer.data(), buffer.capacity());
+            if (nCharsRead = ifile.gcount(); nCharsRead > 0)
             {
-                ZeroMemory(buffer, 4096);
-                _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-                if (nCharsRead > 0)
+                buffer = buffer.data(); 
```

Comment:
the `buffer = buffer.data(); ` part is required. The string in memory has characters written to it but has not been saved to the object. We get the best of both worlds this way.  The size is 0 despite the capacity not being 0. It's also not nested as a selection statement because it's used by more than one scope, should be obvious but some people just don't know.

If you don't think it's not required try it for yourself.